### PR TITLE
Add scalyr

### PR DESCRIPTION
**Pull request by pierreozoux**

### CODE REVIEW

File: stable/scalyr/values.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Default values for scalyr.
+image:
+  repository: scalyr/scalyr-docker-agent
+  tag: latest
```
Comment (unguiculus) [inline stable/scalyr/values.yaml]: Use a fixed version.

File: stable/scalyr/templates/daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{{- if .Values.scalyr.apiKey }}
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```
Comment (unguiculus) [inline stable/scalyr/templates/daemonset.yaml]: Add missing labels. See https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/labels.md.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by pierreozoux** [comment]
Thanks @unguiculus I addressed your comments.

File: stable/scalyr/templates/cm-agent.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "configuration.fullname" . }}
```
Comment (unguiculus) [inline stable/scalyr/templates/cm-agent.yaml]: Is there a reason why you name the config map differently? I'd suggest to use `{{ template "fullname" . }}` as well.

File: stable/scalyr/templates/secrets.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+type: Opaque
+data:
+  api-key: {{ default "MISSING" .Values.scalyr.apiKey | b64enc | quote }}
```
Comment (unguiculus) [inline stable/scalyr/templates/secrets.yaml]: I don't know what makes more sense to you, but are you aware of the new `required` function? See https://github.com/kubernetes/helm/blob/master/docs/charts_tips_and_tricks.md#know-your-template-functions.

File: stable/scalyr/values.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  tag: 2.0.14
+  pullPolicy: IfNotPresent
+
+scalyr:
+  ## You'll need to set this to your Scalyr API key before the agent will run.
+  ## ref: https://www.scalyr.com
+  ##
+  # apiKey:
+
+resources:
```
Comment (unguiculus) [inline stable/scalyr/values.yaml]: Make resources configurable but do not specify defaults. This should be a conscious choice of the user.

File: stable/scalyr/templates/daemonset.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
+{{- if .Values.scalyr.apiKey }}
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "fullname" . }}
```
Comment (unguiculus) [inline stable/scalyr/templates/daemonset.yaml]: `{{ template "name" . }}` would be a better choice for the `app` label. I'd suggest you change it everywhere.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks for the quick changes. I've found a few more things.

**Comment by unguiculus** [comment]
I just tried to deploy the chart and got the following error:
```
$ helm install --name foo stable/scalyr
Error: render error in "scalyr/templates/NOTES.txt": template: scalyr/templates/NOTES.txt:1:14: executing "scalyr/templates/NOTES.txt" at <.Values.scalyr.apiKe...>: can't evaluate field apiKey in type interface {}
```
The same happens with `helm template stable/scalyr`.

**Comment by prydonius** [comment]
@pierreozoux ping, can you update to address @unguiculus's comments?

**Comment by pierreozoux** [comment]
@prydonius can we keep it on hold, I'm currently working for a client, and I want them to sign the CLA before continuing this task.

**Comment by pierreozoux** [comment]
Addressed the comments thanks.
(And my current employer signed the CLA)

File: stable/scalyr/templates/daemonset.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  labels:
+    app: {{ template "name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    heritage: "{{ .Release.Service }}"
+    release: "{{ .Release.Name }}"
+spec:
+  template:
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}
```
Comment (unguiculus) [inline stable/scalyr/templates/daemonset.yaml:15]: Add release label: `release: "{{ .Release.Name }}"`

**Review by unguiculus — CHANGES_REQUESTED** [review]
Just found one more thing.

File: stable/scalyr/templates/daemonset.yaml (hunk 7 above)
Comment (pierreozoux) [reply stable/scalyr/templates/daemonset.yaml:15]: Ok, done

**Review by unguiculus — APPROVED** [review]

**Comment by unguiculus** [comment]
@k8s-bot ok to test

**Comment by unguiculus** [comment]
Sorry for the delay. It should be possible to run install the chart with defaults.

**Comment by unguiculus** [comment]
Closing as stale. Feel free to reopen if/when you decide to work on this again.

**Comment by scottrigby** [comment]
@pierreozoux I drew from some of your earlier work for a new Scalyr Helm chart. Check out https://github.com/scalyr/scalyr-agent-2/issues/118#issuecomment-490135304 if you're still interested in this - always up for collaboration 😄

**Pull request closed without merging**